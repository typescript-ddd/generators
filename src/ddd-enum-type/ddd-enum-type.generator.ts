import { getTemplate } from "../shared/templates";
import { EnumTypeTemplateData } from "./ddd-enum-type.types";

export const generateEnumType = (data: EnumTypeTemplateData) => {
  const template = getEnumTypeTemplate();
  return template(data);
};

export const getEnumTypeTemplate = () =>
  getTemplate("enum-type.hbs"
  ) as HandlebarsTemplateDelegate<EnumTypeTemplateData>;
