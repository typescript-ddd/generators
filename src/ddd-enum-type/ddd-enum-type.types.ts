export interface EnumTypeTemplateData {
    typeName: string;
    description: string;
    values: { key: string; value: string, description?: string }[];
}