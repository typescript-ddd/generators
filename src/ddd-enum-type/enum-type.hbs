/**
 * {{description}}
 * @typedef {{pascal typeName}}
 * @type {object}
 {{#each values}}
 * @property {string} {{key}} - {{aOrAn value}} {{natural value}} {{singular (natural ../typeName)}}.
 {{/each}}
 */
export const {{pascal typeName}} = {
    {{#each values}}
    {{name}}: "{{name}}",
    {{/each}}
};

/**
 * {{description}}
 */
export type {{pascal typeName}} = typeof {{pascal typeName}}[keyof typeof {{pascal typeName}}];