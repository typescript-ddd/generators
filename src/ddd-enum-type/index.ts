export * from "./ddd-enum-type.generator";
export * from "./ddd-enum-type.types";