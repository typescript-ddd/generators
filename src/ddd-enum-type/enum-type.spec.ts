import { generateEnumType } from "./ddd-enum-type.generator";
import { EnumTypeTemplateData } from "./ddd-enum-type.types";

describe("EnumType", () => {
    
      it("should render template", () => {
     // Arrange
     const data: EnumTypeTemplateData = {
        typeName: "Roles",
        description: "Represents the application user roles.",
        values: [
            { key: "Admin", value: "Admin", description: "Admin role" },
            { key: "User", value: "User", description: "User role" },
        ],
     };
    
     // Act
     const result = generateEnumType(data);
    
     // Assert
     expect(result).toMatchSnapshot();
      });
    
    });