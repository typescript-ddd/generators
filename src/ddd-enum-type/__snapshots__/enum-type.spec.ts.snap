// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnumType should render template 1`] = `
"/**
 * Represents the application user roles.
 * @typedef Roles
 * @type {object}
 * @property {string} Admin - an admin role.
 * @property {string} User - a user role.
 */
export const Roles = {
    : "",
    : "",
};

/**
 * Represents the application user roles.
 */
export type Roles = typeof Roles[keyof typeof Roles];"
`;
